tests/e2e/test_install_sh.py

```diff
@@ -0,0 +1,103 @@
+# pylint: disable=too-many-arguments
+import os
+from pathlib import Path
+from shutil import copytree
+
+import pexpect
+import pytest
+
+PROTOSTAR_REPO_URL = "https://github.com/software-mansion/protostar"
+
+
+class ScriptTestingHarness:
+    def __init__(self, process: pexpect.spawn) -> None:
+        self._process = process
+
+    def expect(self, response: str) -> None:
+        self._process.expect(response)
+
+    def expect_kernel_name_uname_prompt(self) -> None:
+        self.expect("\\[uname -s]:")
+
+    def expect_latest_release_response_curl_prompt(self) -> None:
+        self.expect(
+            f"\\[curl -L -s -H Accept: application/json {PROTOSTAR_REPO_URL}/releases/latest]:"
+        )
+
+    def expect_download_curl_prompt(self, filename: str, version: str) -> None:
+        self.expect(
+            f"\\[curl -L {PROTOSTAR_REPO_URL}/releases/download/v{version}/{filename}]:"
+        )
+
+    def expect_tar_info(self, data: str) -> None:
+        self.expect(f"\\[tar {data}]")
+
+    def expect_detected_shell(self, shell_name: str) -> None:
+        self.expect(f"Detected your preferred shell is {shell_name}")
+
+    def send(self, value: str) -> None:
+        self._process.sendline(value)
+
+
+def run_testing_harness(home_path: Path, shell: str) -> ScriptTestingHarness:
+    command = f"bash ./install__testing_harness.sh {home_path} {shell}"
```

Comment:
```suggestion
    command = f"bash ./install_testing_harness.sh {home_path} {shell}"
```

Why does this file have duplicated underscore in its name?

Reply:
The name would suggest the script installs a testing harness. I didn't come up with a better name.

Reply:
I did understood from the context that this is a testing harness for the install script hence I find this underscore superfluous. But this is nitpick which doesn't block this code from merging.

Reply:
Renamed